fri/src/stone_domain.rs

```diff
@@ -0,0 +1,89 @@
+use ark_ff::FftField;
+use ark_poly::{EvaluationDomain, Radix2EvaluationDomain};
+
+// change order of elements in domain
+pub fn change_order_of_elements_in_domain<F: FftField>(elements: &[F]) -> Vec<F> {
+    assert!(elements.len().is_power_of_two());
+    let mut new_elements = vec![F::zero(); elements.len()];
+    let log_len = elements.len().trailing_zeros();
+    let mapping_vec = (0..log_len)
+        .map(|i| (1 << (log_len - 1 - i)))
+        .collect::<Vec<usize>>();
+
+    // [1, 2, 4]
+    // consider array as binary number basis
+    // 0 -> 000 -> 0
+    // 1 -> 001 -> 4
+    // 2 -> 010 -> 2
+    // 3 -> 011 -> 6
+    // 4 -> 100 -> 1
+    // 5 -> 101 -> 5
+    // 6 -> 110 -> 3
+    // 7 -> 111 -> 7
+    for (i, element) in elements.iter().enumerate() {
+        let mut new_index = 0;
+        let mut index = i;
+        for base in mapping_vec.iter() {
+            if index & 1 != 0 {
+                new_index += base;
+            }
+            index >>= 1;
+        }
+        new_elements[new_index] = *element;
+    }
+
+    new_elements
+}
+
+#[allow(dead_code)]
+pub fn make_fft_domains<F: FftField>(
+    domain_size_log: usize,
+    offset: F,
+) -> Vec<Radix2EvaluationDomain<F>> {
+    let mut current_offset = offset;
+    let mut domains = Vec::with_capacity(domain_size_log + 1);
+    for i in (0..=domain_size_log).rev() {
+        let domain = Radix2EvaluationDomain::<F>::new(1 << i)
+            .unwrap()
+            .get_coset(current_offset)
+            .unwrap();
+        domains.push(domain);
+        current_offset = current_offset * current_offset;
```

Comment:
In Stone, when creating domains for each layer, the offset is squared

https://github.com/starkware-libs/stone-prover/blob/ba85f37de3f8473525b7dc03de36cdf8f1567775/src/starkware/fft_utils/fft_bases.inl#L57